index.d.ts

```diff
@@ -284,12 +285,13 @@ declare module '@primer/components' {
     label?: string
   }
 
-  export interface UnderlineNavItemProps extends CommonProps {
+  export interface UnderlineNavLinkProps extends CommonProps {
+    href: string
```

Comment:
```suggestion
    href?: string
```
Should `href` be optional?

Reach Router's Link component accepts a `to` prop instead of `href`. If someone does something like this:
```
<UnderlineNav.Link as={ReachLink} to="#" />
```
will it cause a TypeError?

Reply:
Yep, I think it will. I was going for consistency with ```LinkProps``` where it is also required.

  ```
export interface LinkProps extends CommonProps, TypographyProps {
    href: string
    underline?: boolean
  }
```
Do we want that to be optional as well, in case someone wanted to do:

```<Link as={ReachLink} to="#" />```
?

Reply:
Yeah, that's good idea.